addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -4383,6 +4385,40 @@ export class OdooEditor extends EventTarget {
             this.deselectTable();
         }
         const anchorNode = selection.anchorNode;
+        const isSelectionInEditable = this.isSelectionInEditable(selection);
+        if (this._isMouseStateActive && selection.isCollapsed && isSelectionInEditable) {
+            delete this._isMouseStateActive;
+            const { startContainer, startOffset, endContainer, endOffset } = getDeepRange(this.editable);
+            const linkElement = closestElement(startContainer, "a");
+            if (
+                linkElement &&
+                linkElement.textContent.startsWith("\uFEFF") &&
+                linkElement.textContent.endsWith("\uFEFF")
+            ) {
+                const block = closestBlock(linkElement);
+                const linkIndex = [...block.childNodes].indexOf(linkElement);
```

Comment:
Move these two variables inside the if below since it is the only place that they are used.